### PR TITLE
Add Custom Elements v1 typings

### PR DESCRIPTION
**Pull request by vvakame**
related https://github.com/DefinitelyTyped/DefinitelyTyped/pull/13045

https://w3c.github.io/webcomponents/spec/custom/
https://dom.spec.whatwg.org/#dom-document-createelement

### CODE REVIEW

File: baselines/dom.generated.d.ts [hunk 1]
```diff
@@ -13009,10 +13009,14 @@ interface ElementDefinitionOptions {
     extends: string;
 }
 
+interface ElementCreationOptions {
+    is: string;
```
Comment (rictic) [inline baselines/dom.generated.d.ts]: It looks like `is` is optional as I read the spec

Comment (vvakame) [inline baselines/dom.generated.d.ts]: https://dom.spec.whatwg.org/#dictdef-elementcreationoptions
I think `is` is not optional. 
I want to check document that you seen.

> `[NewObject] Element createElement(DOMString localName, optional ElementCreationOptions options);`

options is optional.

**Comment by vvakame** [comment]
oh, I read README.md again. new guideline added.
https://github.com/Microsoft/TSJS-lib-generator#when-should-a-dom-api-be-included-here

This PR contains in living standard already.
https://dom.spec.whatwg.org/#dom-document-createelement

File: baselines/dom.generated.d.ts (hunk 1 above)
Comment (rictic) [inline baselines/dom.generated.d.ts]: Fair point. I was going off of the listed algorithm, which handles the case where `is` isn't provided. Looking at chrome's implementation confirms your reading, it doesn't crash, but it does `setAttribute('is', '[object Object]')`.

In short, I withdraw the suggestion, `is` should be required.

**Comment by vvakame** [comment]
👀

File: baselines/dom.generated.d.ts (hunk 1 above)
Comment (SaschaNaz) [reply baselines/dom.generated.d.ts]: Months later, but `is` is optional. Web IDL spec says a dictionary member is by default optional unless it has `required` keyword.

Comment (vvakame) [reply baselines/dom.generated.d.ts]: Ah, you are right. I'll fix this PR.
https://heycam.github.io/webidl/#prod-DictionaryMember

**Comment by vvakame** [comment]
I did rebase to master branch & address comment.
@mhegazy could you review this PR?

**Comment by saschanaz** [comment]
This looks simple and short enough to review... Hmm.

**Pull request merged**